// @farcaster/frame-core/dist/schemas/embeds.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.safeParseFrameEmbed = exports.frameEmbedNextSchema = exports.buttonSchema = exports.actionSchema = exports.actionLaunchFrameSchema = void 0;
const zod_1 = require("zod");
const shared_1 = require("./shared");
exports.actionLaunchFrameSchema = zod_1.z.object({
    type: zod_1.z.literal("launch_frame"),
    name: shared_1.frameNameSchema,
    url: shared_1.secureUrlSchema,
    splashImageUrl: shared_1.secureUrlSchema.optional(),
    splashBackgroundColor: shared_1.hexColorSchema.optional(),
});
exports.actionSchema = zod_1.z.discriminatedUnion("type", [
    exports.actionLaunchFrameSchema,
]);
exports.buttonSchema = zod_1.z.object({
    title: shared_1.buttonTitleSchema,
    action: exports.actionSchema,
});
exports.frameEmbedNextSchema = zod_1.z.object({
    version: zod_1.z.literal("next"),
    imageUrl: shared_1.secureUrlSchema,
    button: exports.buttonSchema,
});
const safeParseFrameEmbed = (rawResponse) => exports.frameEmbedNextSchema.safeParse(rawResponse);
exports.safeParseFrameEmbed = safeParseFrameEmbed;
