// @farcaster/frame-core/dist/schemas/shared.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.jsonFarcasterSignatureHeaderSchema = exports.encodedJsonFarcasterSignatureSchema = exports.hexColorSchema = exports.buttonTitleSchema = exports.frameNameSchema = exports.secureUrlSchema = void 0;
const zod_1 = require("zod");
exports.secureUrlSchema = zod_1.z
    .string()
    .url()
    .startsWith("https://", { message: "Must be an https url" })
    .max(512);
exports.frameNameSchema = zod_1.z.string().max(32);
exports.buttonTitleSchema = zod_1.z.string().max(32);
exports.hexColorSchema = zod_1.z
    .string()
    .regex(/^#([0-9A-F]{3}|[0-9A-F]{6})$/i, {
    message: "Invalid hex color code. It should be in the format #RRGGBB or #RGB.",
});
exports.encodedJsonFarcasterSignatureSchema = zod_1.z.object({
    header: zod_1.z.string(),
    payload: zod_1.z.string(),
    signature: zod_1.z.string(),
});
exports.jsonFarcasterSignatureHeaderSchema = zod_1.z.object({
    fid: zod_1.z.number(),
    type: zod_1.z.literal("app_key"),
    key: zod_1.z.string().startsWith("0x"),
});
